### ISSUE TITLE
`msvc` should not be used in conan v1?

### ISSUE THREAD

**Issue by maksim-petukhov** [conan-io/docs#3280]
I mean it doesn't work properly. I tried and I failed because of that issue https://github.com/conan-io/conan/issues/12858 (I suppose it is `won't do`?)
conan v1 has `msvc` compiler in settings.yaml so users (me in particular) expect it to work but it's not. It would be great to mention this in the yaml file with a comment like `# please don't use it in v1` and then in docs somewhere. A large error in conan client would be great too because now it's just a lot of `unresolved external symbol` and you're lucky if you notice one `ignoring unknown option '/dynamic'` warning.

**Comment by memsharded** [comment]
I think the issue in https://github.com/conan-io/conan/issues/12858 is a different thing, because that is related to many legacy recipes in ConanCenter that hadn't been updated to be 2.0-compatible. 

But it doesn't mean that the ``msvc`` compiler setting couldn't be used in your own recipes, you can check that it works with:

```bash
# Create a project template
$ conan new mypkg/0.1 -m=cmake_lib

# Now create with the msvc compiler setting
$ conan create . -s compiler=msvc -s compiler.version=192 -s compiler.runtime=dynamic -s compiler.runtime_type=Release -s compiler.cppstd=14

# Check that it is there
$ conan search mypkg/0.1@
Existing packages for recipe mypkg/0.1:

    Package_ID: dfa0038e9a228bccd11e6c518b8fd895174eba88
        [options]
            shared: False
        [settings]
            arch: x86_64
            build_type: Release
            compiler: msvc
            compiler.cppstd: 14
            compiler.runtime: dynamic
            compiler.runtime_type: Release
            compiler.version: 192
            os: Windows
        Outdated from recipe: False
```

Conan settings is not only about recipes in ConanCenter. A large part of Conan users do not use packages from ConanCenter.

**Comment by maksim-petukhov** [comment]
@memsharded
I'm not sure why they need to be 2.0 compatible to work in conan v1. But OK then. In my case I tried to `create` **icu** package (https://github.com/conan-io/conan-center-index/tree/master/recipes/icu/all) in conan 1.59. And it is conan v2 ready (https://github.com/conan-io/conan-center-index/blob/10658342a6999c03616d1ed4dc3666a9c6f1ef62/.c3i/conan_v2_ready_references.yml#L315). My profile is:
```
[settings]
arch=x86
build_type=Debug
compiler=msvc
compiler.cppstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=192
os=Windows
```
And everything is fine until the test_package stage:
```
icu/73.2 (test package): Generator cmake created conanbuildinfo.cmake
icu/73.2 (test package): Generator cmake_find_package created FindICU.cmake
icu/73.2 (test package): Generator txt created conanbuildinfo.txt
icu/73.2 (test package): Aggregating env generators
icu/73.2 (test package): Generated conaninfo.txt
icu/73.2 (test package): Generated graphinfo
Using lockfile: 'C:\b\THIRDPARTY-ICU-JOB2\recipes\icu\all\test_v1_package\build\c8309280574ed5b1d730346b232df247fae0e558/conan.lock'
Using cached profile from lockfile
icu/73.2 (test package): Calling build()

----Running------
> cd C:\b\THIRDPARTY-ICU-JOB2\recipes\icu\all\test_v1_package\build\c8309280574ed5b1d730346b232df247fae0e558 && cmake -G "Visual Studio 16 2019" -DCONAN_LINK_RUNTIME="/dynamic" -DCONAN_CMAKE_CXX_STANDARD="17" -DCONAN_CMAKE_CXX_EXTENSIONS="OFF" -DCONAN_STD_CXX_FLAG="/std:c++17" -DCONAN_IN_LOCAL_CACHE="OFF" -DCONAN_COMPILER="msvc" -DCONAN_COMPILER_VERSION="192" -DCMAKE_INSTALL_PREFIX="C:\b\THIRDPARTY-ICU-JOB2\recipes\icu\all\test_v1_package\build\c8309280574ed5b1d730346b232df247fae0e558\package" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCMAKE_MODULE_PATH="C:/b/THIRDPARTY-ICU-JOB2/recipes/icu/all/test_v1_package/build/c8309280574ed5b1d730346b232df247fae0e558" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -Wno-dev C:\b\THIRDPARTY-ICU-JOB2\recipes\icu\all\test_v1_package

...

----Running------
> cmake --build C:\b\THIRDPARTY-ICU-JOB2\recipes\icu\all\test_v1_package\build\c8309280574ed5b1d730346b232df247fae0e558 --config Debug -- /m:12 /verbosity:minimal
-----------------
Microsoft (R) Build Engine версии 16.11.2+f32259642 для .NET Framework
(C) Корпорация Майкрософт (Microsoft Corporation). Все права защищены.

Checking Build System
  Building Custom Rule C:/b/THIRDPARTY-ICU-JOB2/recipes/icu/all/test_package/CMakeLists.txt
cl : command line warning D9002: ignoring unknown option '/dynamic' [C:\b\THIRDPARTY-ICU-JOB2\recipes\icu\all\test_v1_package\build\c8309280574ed5b1d730346b232df247fae0e558\test_package\test_package.vcxproj]
  test_package.cpp

...

sicuucd.lib(ucase.ao) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '2' doesn't match value '0' in test_package.obj [C:\b\THIRDPARTY-ICU-JOB2\recipes\icu\all\test_v1_package\build\c8309280574ed5b1d730346b232df247fae0e558\test_package\test_package.vcxproj]
...

sicuucd.lib(characterproperties.ao) : error LNK2001: unresolved external symbol __imp__invalid_parameter [C:\b\THIRDPARTY-ICU-JOB2\recipes\icu\all\test_v1_package\build\c8309280574ed5b1d730346b232df247fae0e558\test_package\test_package.vcxproj]

...

a bunch of `error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL'` and `error LNK2001: unresolved external symbol` there

```

You can see `CONAN_LINK_RUNTIME="/dynamic"` and then the warning about unknown option. The same thing we have in https://github.com/conan-io/conan/issues/12858. And if we change the profile to use `Visual Studio` as a compiler than everything works just fine.

Summing up:
1. The recipe is conan v2 ready
2. conan version is 1.59
3. `msvc` in profile
4. building test_package (which is v1) fails because of invalid value of `CONAN_LINK_RUNTIME`

`msvc` can be used in recipes, yes, but it doesn't work properly in conan v1 as you can see.

This is either a docs issue or a conan issue. Please fill free to move this issue to an appropriate repository.

**Comment by maksim-petukhov** [comment]
@memsharded But I don't use conan 2.0 and do not use any 2.0-ready features. We're using `legacy` generator `cmake_find_package` in all our projects and cannot upgrade right now. That's why I'm using `test_v1_package`. Is `msvc` a conan v2 feature?

**Comment by memsharded** [comment]
> We're using legacy generator cmake_find_package in all our projects and cannot upgrade right now

That is fine. If you are still using ``cmake_find_package``, which has been superseded by ``CMakeDeps`` (back in 20-Jan-2021 it was introduced), then you cannot use the ``msvc``.

But that doesn't mean that ``msvc`` should not be used in Conan v1. It can definitely be used in Conan v1. What needs to be dropped from Conan v1 is the usage of ``cmake_find_package`` and ``cmake`` generators.

> Is msvc a conan v2 feature?

No, as I demonstrated above, it is a feature that works in Conan v1. But not together with some v1 legacy generators that have been superseded by new CMakeDeps and CMakeToolchain time ago.
